packages/edge/src/__tests__/lib/commands/drivers-util.test.ts

```diff
@@ -432,105 +443,161 @@ describe('drivers-util', () => {
-			}
+				expect(command.logger.warn).toBeCalledWith('hub record found without locationId', hub)
+			})
+		})
 
-			getLocationsMock
-				.mockResolvedValueOnce(location)
-				.mockResolvedValueOnce(sharedLocation)
-				.mockResolvedValueOnce(nullAllowedLocation)
-				.mockResolvedValueOnce(undefinedAllowedLocation)
+		describe('defaultConfig', () => {
+			test('getItem uses devices.get', async () => {
+				chooseOptionsWithDefaultsMock.mockReturnValueOnce({ allowIndex: false, useConfigDefault: true } as ChooseOptions)
+				selectFromListMock.mockImplementation(async () => 'chosen-hub-id')
 
-			const listItems = selectFromListMock.mock.calls[0][2].listItems
+				expect(await chooseHub(command, 'prompt message', undefined,
+					{ useConfigDefault: true })).toBe('chosen-hub-id')
 
-			expect(await listItems()).toStrictEqual([ownedHub])
+				const defaultValue = selectFromListMock.mock.calls[0][2].defaultValue
 
-			expect(getLocationsMock).toBeCalledTimes(4)
-			expect(command.logger.warn).toBeCalledWith('filtering out location', sharedLocation)
-			expect(command.logger.warn).toBeCalledWith('filtering out location', nullAllowedLocation)
-			expect(command.logger.warn).toBeCalledWith('filtering out location', undefinedAllowedLocation)
-		})
+				expect(defaultValue).toBeDefined()
+				const getItem = defaultValue?.getItem as (id: string) => Promise<Device>
+				expect(getItem).toBeDefined()
+				getDevicesMock.mockResolvedValueOnce(hub)
 
-		test('list function warns when hub does not have locationId', async () => {
-			chooseOptionsWithDefaultsMock.mockReturnValueOnce({ allowIndex: false } as ChooseOptions)
-			selectFromListMock.mockImplementation(async () => 'chosen-hub-id')
+				expect(await getItem('id-to-check')).toBe(hub)
 
-			expect(await chooseHub(command, 'prompt message', 'command-line-hub-id')).toBe('chosen-hub-id')
+				expect(getDevicesMock).toHaveBeenCalledTimes(1)
+				expect(getDevicesMock).toHaveBeenCalledWith('id-to-check')
+			})
 
-			const hub = { name: 'Hub' } as Device
+			test('userMessage returns expected message', async () => {
+				chooseOptionsWithDefaultsMock.mockReturnValueOnce({ allowIndex: false, useConfigDefault: true } as ChooseOptions)
+				selectFromListMock.mockImplementation(async () => 'chosen-hub-id')
 
-			listDevicesMock.mockResolvedValueOnce([hub])
+				expect(await chooseHub(command, 'prompt message', undefined,
+					{ useConfigDefault: true })).toBe('chosen-hub-id')
 
-			const listItems = selectFromListMock.mock.calls[0][2].listItems
+				const defaultValue = selectFromListMock.mock.calls[0][2].defaultValue
+
+				expect(defaultValue).toBeDefined()
+				const userMessage = defaultValue?.userMessage as (hub: Device) => string
+				expect(userMessage).toBeDefined()
+
+				expect(userMessage(hub)).toBe('using previously specified default hub labeled "hub label" (hub-device-id)')
+			})
+		})
+
+		test('defaultConfig.userMessage return valid message', async () => {
```

Comment:
Is this a duplicate of the previous test?

Reply:
Another good catch. Removed.